api/core/rag/rerank/weight_rerank.py

```diff
@@ -37,11 +37,10 @@ def run(
         :return:
         """
         unique_documents = []
-        doc_id = set()
+        doc_ids = set()
         for document in documents:
-            doc_id = document.metadata.get("doc_id")
-            if doc_id not in doc_id:
-                doc_id.add(doc_id)
+            if document.metadata["doc_id"] not in doc_ids:
+                doc_ids.add(document.metadata["doc_id"])
                 unique_documents.append(document)
```

Comment:
Here's another optimised version in single-line code. ✌🏻

`unique_documents = {doc.metadata["doc_id"]: doc for doc in documents}.values()`

Reply:
in rerank_model.py which dealt with document.provider == "external"
```
elif document.provider == "external":
    if document not in unique_documents:
        unique_documents.append(document)
```
is this the only way to judge whether a external document duplicate?
does a external document has a doc_id in metadata?

Reply:
Just a piece of advice. We could ignore it.
LGTM for now.